File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    i = 0
    stk = []
    while True:
        if i >= len(arr):
            return stk if stk else [-1]
        if not stk:
            stk.append(arr[i])
            i += 1
        elif stk[-1] == arr[i]:
            stk.pop()
            i += 1
        else:
            stk.append(arr[i])
            i += 1

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        List<Integer> stk = new ArrayList<>();
        int i = 0;
        while(i < arr.length) {
                if (stk.isEmpty()) {
                    stk.add(arr[i]);
                    i++;
                } else if (stk.get(stk.size() - 1) == arr[i]) {
                    stk.remove(stk.size() - 1);
                    i++;
                } else {
                    stk.add(arr[i]);
                    i++;
                }
        }
        
        int[] answer = stk.stream().mapToInt(j -> j).toArray();
        return answer.length == 0 ? new int[]{-1} : answer;
    }
}